Support for Swift 3 (and Xcode 9.2 beta 2) 🎉

freak4pc opened the pull request:
Aimed to resolve #44 and similar reports.

I’ve migrated the entire library and tests look good on my end. 

I’ve also tested with latest version of Xcode 9 beta 2 with no compilation issues and all tests running.

Let me know if any questions.

Thanks!

esttorhe commented:
You are amazing @freak4pc !!!

As soon as the `CI` is 💚  I'll merge.

Would you like to be invited to the org?

freak4pc commented:
I'm already part of RxSwiftCommunity :)

esttorhe commented:
Ugh! Sorry, obviously I'm not too good at reading 😳 🙈

esttorhe commented on RxViewModel.podspec:
```diff
@@ -1,7 +1,7 @@
 Pod::Spec.new do |s|
   s.name             = "RxViewModel"
-  s.version          = "3.1.4"
+  s.version          = "3.2.0"
```
My only comment would be to bump this to `4.0.0` since it depends on `RxSwift` `3.4` which adds some breaking changes; that way we can keep `semver` working correctly for anyone using the app.

freak4pc replied:
Makes total sense. Let me fix that up real quick :)

freak4pc commented:
Travis is causing some issues with SwiftLint. Hopefully - almost there :)

esttorhe commented:
Yeah I noticed; I think the images come now provisioned with `SwiftLint` and don't need to install it manually.

I was going to remove the install step directly on `master` as well as bump the image to `Xcode8.3` but decided to wait and submit a `PR` in case the image brakes everything (although I highly doubt it)

freak4pc commented:
@esttorhe Actually will probably want to bump Xcode as part of this as well as I'm not sure this will correctly build on Xcode 7.3 anyways 🤔 

Anyways do we want Xcode 8.3 or [any of the older versions](https://docs.travis-ci.com/user/osx-ci-environment/#OS-X-Version) ?

freak4pc commented:
Ok i've given up on having the latest of SwiftLint, plus it is insanely slow to update HomeBrew at that point. I removed it since it seems like the better decision and also updated the osx_image to Xcode 8.3.

freak4pc commented:
Phewww, Seems like all should be green now ! 💚 
That was a toughie. A lot of old dependencies etc etc. 
🎉🎉🎉🎉🎉🎉🎉🎉🎉

`push` cleared out so I imagine `PR` clears out as well but lets hang tight.

There are just two Swiftlint/Danger warnings there which we can fix but I don't think its a big deal right now. I also think its kinda odd since it's not in the "Source" folder which should be the only one included. 🤔 

Regardless this looks good to me to merge and push a new version out, All you! @esttorhe

freak4pc commented:
All green! 🤘 
4.0.0 can go up 💯

esttorhe commented:
woohooo!!!
You are a hero @freak4pc !
